layers/stp.go

```diff
@@ -7,21 +7,149 @@
 package layers
 
 import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"net"
+
 	"github.com/google/gopacket"
 )
 
+type SWITCHID struct {
+	Priority uint16 // Bridge priority
+	SysID    uint16 // VLAN ID
+	HwAddr   net.HardwareAddr
+}
+
 // STP decode spanning tree protocol packets to transport BPDU (bridge protocol data unit) message.
 type STP struct {
 	BaseLayer
+	ProtocolID        uint16
+	Version           uint8
+	Type              uint8
+	TC, TCA           bool // TC: Topologie change ; TCA: Topologie change ack
+	RouteID, BridgeID SWITCHID
+	Cost              uint32
+	PortID            uint16
+	MSG_AGE           uint16
```

Comment:
MsgAge or MessageAge or somesuch?

In short, is there a way that the CAPS_WITH_UNDERSCORE fields can become CamelCase fields?  It's just slightly more Go-y

Reply:
@gconnell thanks for taking time to read the code and giving me a list of idea how to make it better.
i agree with all your suggestions.
i will try to implement them as soon as possible